### ISSUE TITLE
Subsonic api and network errors.

### ISSUE THREAD

**Issue by bozo83** [ampache/ampache#1111]
Hey I've been using Ampache for a little while now and have had it working perfectly both through the webplayer and through Subsonic android apps. (Subsonic music streaming & Dsub)

Ampache is installed on my VPS rather than my local machine. 

Now over night I increased the size of my Ampache library (a little under 20,000 tracks currently). When trying to refresh my folder list in either of the Subsonic apps it just keeps responding with:

        A network error occurred. Retrying 1 of 4
        A network error occurred. Retrying 2 of 4
        A network error occurred. Retrying 3 of 4
        A network error occurred. Retrying 4 of 4

Before eventually failing and displaying this error:

        A network error occurred. Please check the server address or try again later

I've set debug to level 5 and this is what is logged when I try to refresh my folders in subsonic:

        2016-01-25 18:00:47 [Bozo] (session) -> 1bbogrsfrnof3f8hobqsqvadh4 has been extended to Mon, 25 Jan 2016 19:00:47 +0000 extension length 3600 
        2016-01-25 18:00:48 [Bozo] (ajax.server.php) -> Called for page: {index} 

I've entered the following line in subsonic_api.class.php

        set_time_limit(300);

But that doesn't seem to change anything. I've looked at mysql logs and manager and it doesn't appear that it's creating any errors reading the database, nor is the database timing out and causing the issue. 

I've used other ampache clients and they all work fine. They are all loading the complete library. The only reason I would rather use subsonic app is because you can adjust transcoding settings from within the app so it makes it a little easier to control the quality of music as I go between different wifi networks. (my home network is slow so I transcode smaller than I do at work for instance). Plus (no offence, as I know there are a couple of developers for ampache apps on here) Dsub just looks and works so much better.

So, any ideas? Thanks in advance for any help you can provide and if I've missed any information out then I'll do my best to fill in the gaps. 

Thanks again.

**Comment by Afterster** [comment]
For starting, which version are you using?

**Comment by bozo83** [comment]
I knew I'd miss something out. Latest dev branch.

**Comment by Afterster** [comment]
Ok should be fine then or it is a new bug. Could you check Apache logs this time and try to open the requested faulting page directly with your browser?

**Comment by bozo83** [comment]
Hey. Thanks for getting back to me. Sorry about the delay in getting back to you. Subsonic is now loading the library but is now not transcoding. Whereas it was before. 

When I try to transcode a song now I get the following in logs:

```
2016-01-30 14:09:41 [bozo] (play) -> Playing file (/home/punkishd/public_html/bozo/music/uploads/28 Days/28 Days/28 Days - 4 - 1988.mp3}... 
2016-01-30 14:09:41 [bozo] (play) -> Media type {mp3} 
2016-01-30 14:09:41 [bozo] (play) -> Custom play action {} 
2016-01-30 14:09:41 [bozo] (play) -> Transcode to {} 
2016-01-30 14:09:41 [bozo] (play) -> Transcoding because explicit bitrate request 
2016-01-30 14:09:41 [bozo] (stream.class.php) -> Starting transcode for {/home/punkishd/public_html/bozo/music/uploads/28 Days/28 Days/28 Days - 4 - 1988.mp3}. Type {}. Options: Array 
2016-01-30 14:09:41 [bozo] (stream.class.php) -> ( 
2016-01-30 14:09:41 [bozo] (stream.class.php) ->     [bitrate] => 192 
2016-01-30 14:09:41 [bozo] (stream.class.php) -> ) 
2016-01-30 14:09:41 [bozo] (stream.class.php) -> }... 
2016-01-30 14:09:41 [bozo] (media) -> Using default target format 
2016-01-30 14:09:41 [bozo] (media) -> Transcode settings: from mp3 to mp3 
2016-01-30 14:09:41 [bozo] (media) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1 
2016-01-30 14:09:41 [bozo] (stream) -> Final transcode bitrate is 192 
2016-01-30 14:09:41 [bozo] (stream) -> %SAMPLE% not in transcode command 
2016-01-30 14:09:41 [bozo] (stream) -> %MAXBITRATE% not in transcode command 
2016-01-30 14:09:41 [bozo] (stream) -> %RESOLUTION% not in transcode command 
2016-01-30 14:09:41 [bozo] (stream) -> %QUALITY% not in transcode command 
2016-01-30 14:09:41 [bozo] (stream) -> Transcode command: ffmpeg -i '/home/punkishd/public_html/bozo/music/uploads/28 Days/28 Days/28 Days - 4 - 1988.mp3' -vn -b:a 192K -c:a libmp3lame -f mp3 pipe:1 
2016-01-30 14:09:41 [bozo] (stream) -> Transcode command prefix: exec  
2016-01-30 14:09:41 [bozo] (play) -> Starting stream of /home/punkishd/public_html/bozo/music/uploads/28 Days/28 Days/28 Days - 4 - 1988.mp3 with size 3417853 
2016-01-30 14:09:41 [bozo] (play) -> Registering stream stats for {28 Days - 1988}... 
2016-01-30 14:09:41 [bozo] (user.class.php) -> Updating stats for {song/449} {android}... 
2016-01-30 14:09:41 [bozo] (user.class.php) -> Scrobbling explicitly skipped 
2016-01-30 14:09:41 [bozo] (stream) -> Process is not running, kill skipped. 
2016-01-30 14:09:41 [bozo] (play) -> Stream ended at 0 (0) bytes out of  
2016-01-30 14:09:41 [bozo] (Art) -> Local image art /home/punkish/public_html/music/meta/album/86/default/art-original.jpg cannot be read. 
2016-01-30 14:09:41 [bozo] (Art) -> Invalid image passed 
2016-01-30 14:09:41 [bozo] (Art) -> Not trying to generate thumbnail, invalid data passed
```

I don't know if that's to do with the earlier bug or if it's completely separate. Cheers.

**Comment by bozo83** [comment]
OK, so looking into it further. Transcoding isn't working across the whole site. So neither the web player nor the mobile app will transcode. I figured it was an issue with ffmpeg. But avconv isn't transcoding either. My config file seems to be laid out right from what I can see. I have confirmed that ffmpeg IS installed. I just can't figure this one out. I'm gonna quickly do a fresh install and see how that goes.
